ios/Classes/core/PMManager.m

```diff
@@ -440,15 +440,22 @@ - (void)fetchThumb:(PHAsset *)asset option:(PMThumbLoadOption *)option resultHan
                       contentMode:option.contentMode
                           options:requestOptions
                     resultHandler:^(PMImage *result, NSDictionary *info) {
-        BOOL downloadFinished = [PMManager isDownloadFinish:info];
+        if ([handler isReplied]) {
+            return;
+        }
 
-        if (!downloadFinished) {
+        if (info[PHImageErrorKey]) {
+            [handler reply: nil];
+            [self notifySuccess:progressHandler];
```

Comment:
Should be `[self notifyProgress:progressHandler progress:progress state:PMProgressStateFailed];` ?

Reply:
I took a look at the previous, so you basically just changing the order of the flow and adding this predicate. Can you simply add this predicate at the top of the callback?

Reply:
Sure thing, won't that cause any issues if it gets called twice or are we sure it won't get called twice? I changed the order because `if ([handler isReplied])` was preventing multiple calls to `[handler reply: data];` earlier and I wanted to be safe when calling this myself.

Reply:
Ah cool. I like the idea!

Reply:
Ah, I saw just a like and thought it meant that I should change that anyway. But I reverted it now. In this case is there anything you want me to change still or is this fine?

Reply:
Yes please change it. Nothing should be changed if the tests are happy.

Reply:
Looks like some Android tests are not passing but it doesn't seem to be related to my changes.